src/main/java/com/avispl/symphony/dal/amx/muse/communicator/AMXMuseCommunicator.java

```
@@ -0,0 +1,219 @@
+//            }
+//            if (i <= 2) {
+//                statistics.put(String.format("NetworkInterface[%s]#MACAddress", i), extractResponseValue(send(String.format(Constant.Command.NETWORK_INTERFACE_MAC, i).getBytes())));
+//                statistics.put(String.format("NetworkInterface[%s]#Gateway", i), extractResponseValue(send(String.format(Constant.Command.NETWORK_INTERFACE_GATEWAY, i).getBytes())));
+//                statistics.put(String.format("NetworkInterface[%s]#IPAddress", i), extractResponseValue(send(String.format(Constant.Command.NETWORK_INTERFACE_IP_ADDRESS, i).getBytes())));
+//                statistics.put(String.format("NetworkInterface[%s]#DHCP", i), extractResponseValue(send(String.format(Constant.Command.NETWORK_INTERFACE_DHCP, i).getBytes())));
+
+//                for (int j = 1; j <= 3; j++) {
+//                    statistics.put(String.format("NetworkInterface[%s]#DNSServer[s]", i, j), extractResponseValue(send(String.format(Constant.Command.NETWORK_INTERFACE_DNS_SERVER, i, j).getBytes())));
+//                }
+//                statistics.put(String.format("NetworkInterface[%s]#SubnetMask", i), extractResponseValue(send(String.format(Constant.Command.NETWORK_INTERFACE_SUBNET_MASK, i).getBytes())));
+//            }
+//        }
+        disconnect();
+        extendedStatistics.setStatistics(statistics);
+        extendedStatistics.setControllableProperties(controls);
+        return Arrays.asList(extendedStatistics);
+    }
+
+    /**
+     * Extract the response value from device response with format
+     * @get {"path":"$path","value":$value}
+     *
+     * @param response raw byte array response
+     * @return String value of the command response
+     *
+     * @throws Exception if any communication error occurs
+     * */
+    private String extractResponseValue(byte[] response) throws Exception {
+        String strResponse = new String(response);
```

I think we should throw exception if communication fails instead of returning RuntimeException without message